src/com/owncloud/android/widgets/ShortcutsWidgetConfigureActivity.java

```diff
@@ -0,0 +1,146 @@
+/**
+ *   ownCloud Android client application
+ *
+ *   @author masensio
+ *   Copyright (C) 2015 ownCloud Inc.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License version 2,
+ *   as published by the Free Software Foundation.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package com.owncloud.android.widgets;
+
+import android.app.Activity;
+import android.app.Fragment;
+import android.appwidget.AppWidgetManager;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.widget.Toast;
+
+import com.owncloud.android.R;
+import com.owncloud.android.ui.dialog.AccountsDialogFragment;
+
+
+/**
+ * The configuration screen for the {@link ShortcutsWidget ShortcutsWidget} AppWidget.
+ */
+public class ShortcutsWidgetConfigureActivity extends Activity
+        implements AccountsDialogFragment.AccountDialogFragmentListener{
+
+    private int mAppWidgetId = AppWidgetManager.INVALID_APPWIDGET_ID;
+    private static final String PREFS_NAME = "com.owncloud.android.widgets.ShortcutsWidget";
+    private static final String PREF_PREFIX_KEY = "appwidget_";
+
+    public ShortcutsWidgetConfigureActivity() {
+        super();
+    }
+
+    @Override
+    public void onCreate(Bundle icicle) {
+        super.onCreate(icicle);
+
+        // Set the result to CANCELED.  This will cause the widget host to cancel
+        // out of the widget placement if the user presses the back button.
+        setResult(RESULT_CANCELED);
+
+          // Find the widget id from the intent.
+        Intent intent = getIntent();
+        Bundle extras = intent.getExtras();
+        if (extras != null) {
+            mAppWidgetId = extras.getInt(
+                    AppWidgetManager.EXTRA_APPWIDGET_ID, AppWidgetManager.INVALID_APPWIDGET_ID);
+
+        }
+
+        // If this activity was started with an intent without an app widget ID,
+        // finish with an error.
+        if (mAppWidgetId == AppWidgetManager.INVALID_APPWIDGET_ID) {
+            finish();
+            return;
+        }
+
+        // Show dialog with the accounts
+        AccountsDialogFragment accountsDialogFragment = new AccountsDialogFragment().getInstance();
```

Comment:
Wait what? new A.getInstance() ? It's a static method. It will create two objects.